bundles/binding/org.openhab.binding.caldav-command/src/main/java/org/openhab/binding/caldav_command/internal/CalDavBinding.java

```diff
@@ -0,0 +1,359 @@
+		
+		if (this.itemRegistry == null) {
+			logger.error("item registry is not set");
+			return;
+		}
+		
+		this.doAction(event, EventUtils.SCOPE_END);
+		this.handleForEventPreview();
+	}
+	
+	@Override
+	public void calendarReloaded(String calendarId) {
+		if (!readCalendars.contains(calendarId)) {
+			return;
+		}
+		
+		if (this.itemRegistry == null) {
+			logger.error("item registry is not set");
+			return;
+		}
+		
+		if (!this.calendarReloaded) {
+			this.doActionInitial();
+		}
+		this.handleForEventPreview();
+		
+		this.calendarReloaded = true;
+	}
+	
+	private void doActionInitial() {
+		List<CalDavEvent> events = calDavLoader.getEvents(new CalDavQuery(this.readCalendars, DateTime.now(), DateTime.now()));
+		Map<String, EventUtils.EventContent> map = new HashMap<String, EventUtils.EventContent>();
+		for (CalDavEvent calDavEvent : events) {
+			final List<EventUtils.EventContent> parseContent = EventUtils.parseContent(calDavEvent, this.itemRegistry, null);
+			for (EventUtils.EventContent eventContent : parseContent) {
+				if (disabledItems.contains(eventContent.getItem().getName())) {
+					continue;
+				}
+				
+				EventUtils.EventContent currentEventContent = map.get(eventContent.getItem().getName());
+				if (eventContent.getTime().isBefore(DateTime.now())
+						&& (currentEventContent == null || eventContent.getTime().isAfter(currentEventContent.getTime()))) {
+					map.put(eventContent.getItem().getName(), eventContent);
+				}
+			}
+		}
+		
+		for (EventUtils.EventContent currentEventContent : map.values()) {
+			eventPublisher.sendCommand(currentEventContent.getItem().getName(), currentEventContent.getCommand());
+			logger.debug("setting initial value for {} to {}", currentEventContent.getItem().getName(), currentEventContent.getCommand());
+		}
+	}
+	
+	private void doAction(CalDavEvent event, String scope) {
+		final List<EventUtils.EventContent> parseContent = EventUtils.parseContent(event, this.itemRegistry, scope);
+		for (EventUtils.EventContent eventContent : parseContent) {
+			if (disabledItems.contains(eventContent.getItem().getName())) {
+				continue;
+			}
+			
+			eventPublisher.sendCommand(eventContent.getItem().getName(), eventContent.getCommand());
+		}
+	}
+	
+	private void updateItemState(CalDavNextEventConfig config, List<CalDavEvent> events) {
+		String itemName = config.getItemNameToListenTo();
+		String itemNamePreview = config.getItemName();
+		logger.trace("update item state for item: {}", itemName);
+		
+		Command state = null;
+		DateTime time = null;
+		
+		if (calDavLoader == null) {
+			logger.warn("caldav loader is not set");
+			return;
+		}
+		
+		for (CalDavEvent calDavEvent : events) {
+			Item item = null;
+			try {
+				item = this.itemRegistry.getItem(itemName);
+			} catch (ItemNotFoundException e) {
+				logger.error("item {} could not be found", itemName);
+				continue;
+			}
+			
+			final List<EventUtils.EventContent> parseContent = EventUtils.parseContent(calDavEvent, item);
```

Comment:
I change it, but my intention to do it this is way is: I would like to see that the item could not be found in the registry and because of this it has to be skipped. If the try-catch is around of a big block it is not directly visible. But nevertheless i change it.